src/type/schema.js

```diff
@@ -269,3 +278,10 @@ function typeMapReducer(map: TypeMap, type: ?GraphQLType): TypeMap {
 
   return reducedMap;
 }
+
+function getDirectiveArgTypes(directive: GraphQLDirective) {
+  if (!isDirective(directive)) {
```

Comment:
Why do you need this check?

Reply:
There is a test that `validateSchema` throws a specific error when `directives` is invalid (a string instead of a proper instance). `new GraphQLSchema()` does not perform this validation, so for the mentioned test to pass, the constructor needs to accept the invalid directive.

Reply:
@Yogu Make sense 👍